codec_record.go

```diff
@@ -172,9 +172,15 @@ func encoderOfStruct(cfg *frozenConfig, schema Schema, typ reflect2.Type) ValEnc
 		}
 
 		defaultType := reflect2.TypeOf(def)
+
+		defaultEncoder := encoderOfType(cfg, field.Type(), defaultType)
+		if defaultType.LikePtr() {
+			defaultEncoder = &onePtrEncoder{defaultEncoder}
+		}
+
```

Comment:
Nit: This is all the same block as it is a single sequence.

```suggestion
		defaultEncoder := encoderOfType(cfg, field.Type(), defaultType)
		if defaultType.LikePtr() {
			defaultEncoder = &onePtrEncoder{defaultEncoder}
		}
```